### PR TITLE
JSON to Zoho CRM - Products

### LINKED ISSUE #173: JSON to Zoho CRM - Products

### PR DESCRIPTION
**Pull request by jannejjj**
closes #173 

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
  - [x] If applicable, the process name follows the naming convention "{source/format} to {destination/format} - {object}"
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots
- [x] Template parameters checked
  - [x] Parameter follows the CamelCase naming convention, abbreviations included, e.g. ConnectionString, BaseUrl, ApiKey
  - [x] Parameter descriptions are good and end with dots
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output, e.g. tokens in the header.
- [x] Process end points like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
  - [x] Errors are pointed downwards from their respective source, e.g. an exclusive decision
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Template files are going to the correct folder
- [x] Markdown paragraphs are separated with two newlines, so there is a blank line inbetween two paragraphs
- [x] If some system is integrated using e.g. a HTTP task (meaning we do not use a native connector for the system), then the system should be mentioned in additional systems used in metadata.json
- [x] If the template is handling some sample JSON data - make sure that the sample data is included in the long description (just one-two elements is enough)
- [x] If the template is assuming that there is a database table ready for data insertion / updates - include the table structure in the long description as a `CREATE TABLE` scipt.

## Summary by CodeRabbit

- **New Features**
  - Introduced a template for integrating product data from JSON files into Zoho CRM, enabling seamless insert/update operations.
  - Added metadata support for project management, including project name and versioning.
  - Implemented a process template for managing product information with automated error handling and reporting.

- **Documentation**
  - Provided a sample JSON structure and prerequisites for using the new templates, enhancing user guidance and experience.

### CODE REVIEW

**Review by charlotta-frends — CHANGES_REQUESTED** [review]
The process looks clear and works fine! Comments:

- Noticed that "skip logging" is on in all of the Zoho requests, though I think it's enough to have it only in the access token task 
- Does the task "Check if product already exists in Zoho" need an error check and a throw/return after it?

File: Templates/CRMs/Zoho CRM/JSON to Zoho CRM - Products/long-description.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+# Implementation and Usage Notes
+
+This template reads product information from a JSON file and inserts or updates the products into Zoho CRM. To determine if a product should be inserted or updated, the product codes of the existing products on Zoho CRM are compared to the codes of the products to be inserted or updated.
+
+This template assumes that the input JSON structure is correct and uses the field names that the Zoho API expects. The template does not perform any mapping for the JSON data.
+
+The process variables include the regional domain URLs for your Zoho instance, 
+
+**Sample input JSON structure**
```
[the remark this answers is not in the document]
Comment (jannejjj) [reply Templates/CRMs/Zoho CRM/JSON to Zoho CRM - Products/long-description.md:22]: You're right, I'll make the JSON example smaller, only including the most crucial fields.

File: Templates/CRMs/Zoho CRM/JSON to Zoho CRM - Products/long-description.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+- It is known which domain your Zoho instance is in, e.g., EU or America.
+- The Frends agent has access to the folder from which the JSON file should be read.
+
+# Implementation and Usage Notes
+
+This template reads product information from a JSON file and inserts or updates the products into Zoho CRM. To determine if a product should be inserted or updated, the product codes of the existing products on Zoho CRM are compared to the codes of the products to be inserted or updated.
+
+This template assumes that the input JSON structure is correct and uses the field names that the Zoho API expects. The template does not perform any mapping for the JSON data.
+
+The process variables include the regional domain URLs for your Zoho instance,
```
[the remark this answers is not in the document]
Comment (jannejjj) [reply Templates/CRMs/Zoho CRM/JSON to Zoho CRM - Products/long-description.md:20]: Probably, i'll take a look!

**Comment by jannejjj** [comment]
> The process looks clear and works fine! Comments:
> 
> * Noticed that "skip logging" is on in all of the Zoho requests, though I think it's enough to have it only in the access token task

I had left it on because I thought that the OAuth token is logged by default, but obviously it isn't.

> * Does the task "Check if product already exists in Zoho" need an error check and a throw/return after it?

Right, it probably does! Added that.

**Review by charlotta-frends — APPROVED** [review]
All requested fixes done, approved! 👌

**Pull request merged**